lineman/app/models/group_model.coffee

```diff
@@ -148,6 +148,12 @@ angular.module('loomioApp').factory 'GroupModel', (DraftableModel, AppConfig) ->
         @remove()
         _.each @memberships(), (m) -> m.remove()
 
+    markAsRead: =>
+      @remote.postMember(@key, 'mark_as_read')
+      _.map @discussions(), (discussion) ->
+        return unless discussion.isUnread()
+        discussion.update(lastReadAt: moment(), readSalientItemsCount: discussion.salientItemsCount)
```

Comment:
Could be 404 bad network connection, could be you logged out elsewhere, your token expired, we 500... Who knows, servers are weird, heh. I have a strong suspicion we could end up with records where the activityCount is 0, but readCommentCount < commentCount, and that just seems weird and fragile.

I think the solution here is to do `discussion.update(readOnClient: true)` (reads better anyway), and then isUnread becomes
```
isUnread: ->
  @readOnClient or # current isUnread logic
```

Then when Faye updates the record, it'll populate the correct values from the server, and set readOnClient back to false (we'll serialize out readOnClient as false with each discussion)